syntax = "proto3";

package auth;

service AuthService {
  rpc Register (RegisterRequest) returns (AuthResponse);
  rpc Login (LoginRequest) returns (AuthResponse);
  rpc ValidateToken (TokenRequest) returns (ValidateResponse);
}

message RegisterRequest {
  string username = 1;
  string password = 2;
  string email = 3;
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message TokenRequest {
  string token = 1;
}

message AuthResponse {
  string token = 1;
  string message = 2;
}

message ValidateResponse {
  bool is_valid = 1;
}
